src/coreclr/nativeaot/Runtime/unix/PalCreateDump.cpp

```diff
@@ -0,0 +1,572 @@
+            }
+
+            // Add the current thread id to the command line. This function is always called on the crashing thread.
+            crashThreadArg = FormatInt(PlatformGetCurrentThreadId());
+            if (crashThreadArg != nullptr)
+            {
+                argv[argc++] = "--crashthread";
+                argv[argc++] = crashThreadArg;
+            }
+
+            if (siginfo != nullptr)
+            {
+                signalCodeArg = FormatInt(siginfo->si_code);
+                if (signalCodeArg != nullptr)
+                {
+                    argv[argc++] = "--code";
+                    argv[argc++] = signalCodeArg;
+                }
+                signalErrnoArg = FormatInt(siginfo->si_errno);
+                if (signalErrnoArg != nullptr)
+                {
+                    argv[argc++] = "--errno";
+                    argv[argc++] = signalErrnoArg;
+                }
+                signalAddressArg = FormatInt64((uint64_t)siginfo->si_addr);
+                if (signalAddressArg != nullptr)
+                {
+                    argv[argc++] = "--address";
+                    argv[argc++] = signalAddressArg;
+                }
+            }
+
+            argv[argc++] = nullptr;
+            _ASSERTE(argc < MAX_ARGV_ENTRIES);
+        }
+
+        CreateCrashDump(argv, nullptr, 0);
+
+        free(signalArg);
+        free(crashThreadArg);
+        free(signalCodeArg);
+        free(signalErrnoArg);
+        free(signalAddressArg);
+    }
+}
+
+void
+PalCreateCrashDumpIfEnabled()
+{
+    PalCreateCrashDumpIfEnabled(SIGABRT, nullptr);
+}
+
+/*++
+Function:
+  PalGenerateCoreDump
+
+Abstract:
+  Public entry point to create a crash dump of the process.
+
+Parameters:
+    dumpName
+    dumpType:
+        Normal = 1,
+        WithHeap = 2,
+        Triage = 3,
+        Full = 4
+    flags
+        See enum
+
+Return:
+    true success
+    false failed
+--*/
+bool
+PalGenerateCoreDump(
+    const char* dumpName,
+    int dumpType,
+    uint32_t flags,
+    char* errorMessageBuffer,
+    int cbErrorMessageBuffer)
+{
+    const char* argvCreateDump[MAX_ARGV_ENTRIES];
+    if (dumpType < 1 || dumpType > 4)
+    {
+        return false;
+    }
+    if (dumpName != nullptr && dumpName[0] == '\0')
+    {
+        dumpName = nullptr;
+    }
+    bool result = BuildCreateDumpCommandLine(argvCreateDump, dumpName, nullptr, dumpType, flags);
+    if (result)
+    {
+        result = CreateCrashDump(argvCreateDump, errorMessageBuffer, cbErrorMessageBuffer);
+    }
+    return result;
+}
+
+/*++
+Function
+  PalCreateDumpInitialize()
+
+Abstract
+  Initialize the process abort crash dump program file path and
+  name. Doing all of this ahead of time so nothing is allocated
+  or copied in abort/signal handler.
+
+Return
+  true - succeeds, false - fails
+
+--*/
+bool
+PalCreateDumpInitialize()
+{
+    CLRConfigNoCache enabledCfg = CLRConfigNoCache::Get("DbgEnableMiniDump", /*noprefix*/ false, &getenv);
```

Comment:
We have RhConfig mechanism for this, we shouldn't need to bring over a parallel configuration mechanism.

Reply:
That looks good except it only looks for the DOTNET_ prefixed env vars and .NET Core still looks for both DOTNET_ and COMPlus_. 

Is it ok for native aot to look at only DOTNET_ variables?

Reply:
Yep, we don't look for COMPlus for any other configuration options either. DOTNET became the preferred prefix in 6, Native AOT came out in 7. NativeAOT doesn't have the compat requirement for the old prefix because nobody was using it with that one.